session/manager.go

```diff
@@ -181,11 +182,8 @@ func (sm *Manager) UserSession(ctx context.Context) (*types.UserSession, error)
 	err := pc.RetrieveOne(ctx, sm.Reference(), []string{"currentSession"}, &mgr)
 	if err != nil {
 		// It's OK if we can't retrieve properties because we're not authenticated
-		if f, ok := err.(types.HasFault); ok {
-			switch f.Fault().(type) {
-			case *types.NotAuthenticated:
-				return nil, nil
-			}
+		if fault.Is(err, (*types.NotAuthenticated)(nil)) {
```

Comment:
It can be be either (we cover this line in the tests). We have the pattern of using typed nil in the DeviceList helper for example: https://github.com/vmware/govmomi/blob/main/object/virtual_device_list.go#L289
And of course in vim25/types init() functions, just avoiding allocations.
But I don't think we gain much by using typed nil here, and I can see how it'd be confusing, so I'll change it.
See example below where typed nil is not a nil interface:
```go
	f := func(target types.BaseMethodFault) {
		if target == nil {
			log.Printf("target is nil")
			return
		}
		log.Printf("target %T is NOT nil", target)
	}

	f(nil)
	f((*types.NotAuthenticated)(nil))
```

Reply:
I'm glad either works, and I like how it avoids allocations if you want to keep it that way.

Reply:
I think it's easier to read + write how it is, plus the error paths aren't usually hot. We can always optimize later if needed.